core/lib/spree/core/search/base.rb

```diff
@@ -114,6 +115,7 @@ def build_option_value_ids(params)
         end
 
         def prepare(params)
+          @properties[:current_store] = Spree::Store.find_by(id: params[:current_store_id])
```

Comment:
@mad-eel rather than fetching the store again please use `attr_accessor` like it's done for `curent_currency` or `current_user`